server/src/main/java/org/elasticsearch/index/mapper/RootObjectMapper.java

```diff
@@ -527,6 +541,11 @@ private static boolean processField(
         } else if (fieldName.equals("runtime")) {
             if (fieldNode instanceof Map) {
                 Map<String, RuntimeField> fields = RuntimeField.parseRuntimeFields((Map<String, Object>) fieldNode, parserContext, true);
+                if (namespaceValidator != null) {
+                    for (String runtimeField : fields.keySet()) {
+                        namespaceValidator.validateNamespace(null, runtimeField);
```

Comment:
you are doing this twice after calling parseRuntimeFields. Shall we fold that logic into parseRuntimeFields directly>